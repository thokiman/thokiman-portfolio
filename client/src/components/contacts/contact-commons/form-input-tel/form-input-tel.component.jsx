import React from 'react';
import {
  FormInputTelCheckerContainer,
  FormInputTelCheckerIconFalse,
  FormInputTelCheckerIconTrue,
  FormInputTelElement,
  FormInputTelGroup,
  FormLabelElement,
} from './form-input-tel.styles';

export const FormInputTel = ({ handleChange, label, ...props }) => {
  return (
    <FormInputTelGroup className='contact-form-element'>
      {
        <FormInputTelCheckerContainer>
          {props.value.length && props.value.match(RegExp(props.pattern)) ? (
            <FormInputTelCheckerIconTrue />
          ) : (
            <FormInputTelCheckerIconFalse />
          )}
        </FormInputTelCheckerContainer>
      }
      <FormInputTelElement onChange={handleChange} {...props} />
      {label ? (
        <FormLabelElement $formvalue={props.value}>{label}</FormLabelElement>
      ) : null}
    </FormInputTelGroup>
  );
};

export default FormInputTel;
