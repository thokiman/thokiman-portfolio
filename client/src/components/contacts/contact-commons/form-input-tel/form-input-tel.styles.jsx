import styled, { css } from "styled-components";
import { FaCheckCircle, FaTimesCircle } from "react-icons/fa";

const subColor = "#128194";
const mainColor = "#bccfd5";

const shrinkLabel = css`
  top: -15px;
  font-size: 2vw;
  color: ${mainColor};
`;
const shrinkLabelMobile = css`
  @media only screen and (max-width: 800px) {
    color: ${mainColor};
    font-size: 5vw;
    top: -0.5vh;
  }
`;

export const FormInputTelGroup = styled.div`
  position: relative;
  left: 8px;
  margin: 25px 0 20px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;
export const FormInputTelCheckerContainer = styled.span`
  color: ${subColor};
  position: absolute;
  left: -1.5vw;
  @media only screen and (max-width: 800px) {
    left: -3vw;
  }
`;
const FormInputTelCheckerIconStyles = css`
  width: 2vw;
  height: 2vh;
  @media only screen and (max-width: 800px) {
    width: 3.5vw;
    height: 3.5vh;
  }
`;
export const FormInputTelCheckerIconTrue = styled(FaCheckCircle)`
  ${FormInputTelCheckerIconStyles}
`;
export const FormInputTelCheckerIconFalse = styled(FaTimesCircle)`
  ${FormInputTelCheckerIconStyles}
`;

const FormInputTelElementStyles = css`
  background: none;
  position: relative;
  top: 1.5vh;
  left: 1.5vw;
  color: ${subColor};
  font-size: 25px;
  letter-spacing: 0.1em;
  padding: 10px 10px 10px 5px;
  display: block;
  border: none;
  border-radius: 0;
  border-bottom: 1px solid ${subColor};
  margin: 15px 1vw;

  &:-webkit-autofill,
  &:-webkit-autofill:hover,
  &:-webkit-autofill:focus,
  &:-webkit-autofill::first-line {
    -webkit-text-fill-color: ${subColor};
    transition: background-color 6000s ease-in-out;
    font-family: futura-medium;
    font-size: 25px;
  }
  &:focus {
    outline: none;
  }
  &:focus ~ label {
    ${shrinkLabel}
  }
  @media only screen and (max-width: 800px) {
    &:focus ~ label {
      ${shrinkLabelMobile}
    }
  }
`;

const FormInputTelElementAreaCode = css`
  ${FormInputTelElementStyles}
  width: 90px;
`;
const FormInputTelElementPhoneNumber = css`
  ${FormInputTelElementStyles}
  width: 225px;
`;

const getFormInputTelElement = (props) => {
  if (props.name === "area") return FormInputTelElementAreaCode;
  return FormInputTelElementPhoneNumber;
};

export const FormInputTelElement = styled.input`
  ${getFormInputTelElement}
`;

const getFormLabelElementShrink = (props) => {
  if (props.$value.length) return shrinkLabel;
};
const getFormLabelElementShrinkMobile = (props) => {
  if (props.$value.length) return shrinkLabelMobile;
};

export const FormLabelElement = styled.label`
  color: ${subColor};
  font-size: 2vw;
  font-weight: normal;
  pointer-events: none;
  position: absolute;
  top: 1.5vh;
  left: 2vw;
  transition: 300ms ease all;
  ${getFormLabelElementShrink}
  @media only screen and (max-width: 800px) {
    font-size: 6vw;
    top: 2vh;
    left: 5vw;
    ${getFormLabelElementShrinkMobile}
  }
`;

FormInputTelGroup.displayName = "FormInputTelGroup";
FormInputTelCheckerContainer.displayName = "FormInputTelCheckerContainer ";
FormInputTelCheckerIconTrue.displayName = "FormInputTelCheckerIconTrue";
FormInputTelCheckerIconFalse.displayName = "FormInputTelCheckerIconFalse";
FormInputTelElement.displayName = "FormInputTelElement";
FormLabelElement.displayName = "FormLabelElement";
