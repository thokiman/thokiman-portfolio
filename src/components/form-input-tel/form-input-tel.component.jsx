import React from "react";

import {
  FormInputTelGroup,
  FormInputTelCheckerContainer,
  FormInputTelCheckerIconTrue,
  FormInputTelCheckerIconFalse,
  FormInputTelElement,
  FormLabelElement,
} from "./form-input-tel.styles";

const FormInputTel = ({ handleChange, label, ...props }) => {
  return (
    <FormInputTelGroup>
      {
        <FormInputTelCheckerContainer>
          {props.value.length && props.value.match(RegExp(props.pattern)) ? (
            <FormInputTelCheckerIconTrue />
          ) : (
            <FormInputTelCheckerIconFalse />
          )}
        </FormInputTelCheckerContainer>
      }
      <FormInputTelElement onChange={handleChange} {...props} />
      {label ? (
        <FormLabelElement value={props.value}>{label}</FormLabelElement>
      ) : null}
    </FormInputTelGroup>
  );
};

export default FormInputTel;
