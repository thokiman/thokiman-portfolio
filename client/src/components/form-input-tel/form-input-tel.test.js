import React from "react";
import { shallow } from "enzyme";

import { FormInputTel } from "./form-input-tel.component";

describe("FormInputTel component", () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: "phone",
      value: "012345678",
      handleChange: mockHandleChange,
    };

    wrapper = shallow(<FormInputTel {...mockProps} />);
  });

  it("should render FormInputTel component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange method when input changes", () => {
    wrapper.find("FormInputTelElement").simulate("change");

    expect(mockHandleChange).toHaveBeenCalled();
  });

  it("should render FormLabelElement if there is a label", () => {
    expect(wrapper.exists("FormLabelElement")).toBe(true);
  });

  it("should not render FormLabelElement if there is no label", () => {
    const mockNewProps = {
      label: "",
      value: "0123456",
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormInputTel {...mockNewProps} />);

    expect(newWrapper.exists("FormLabelElement")).toBe(false);
  });
});
