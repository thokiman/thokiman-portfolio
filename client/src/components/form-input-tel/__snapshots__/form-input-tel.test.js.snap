// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputTel component should render FormInputTel component 1`] = `ShallowWrapper {}`;
